ArduCopter/autoyaw.cpp

```diff
@@ -266,8 +266,8 @@ float Mode::AutoYaw::yaw_cd()
     default:
         // point towards next waypoint.
         // we don't use wp_bearing because we don't want the copter to turn too much during flight
-        _yaw_angle_cd = copter.pos_control->get_yaw_cd();
-    break;
+        IGNORE_RETURN(copter.pos_control->get_yaw_cd(_yaw_angle_cd));
```

Comment:
I think we need to make sure we initialise `_yaw_angle_cd` correctly in this case, to current yaw angle when starting autoyaw control for example. Otherwise we might endup using a left over yaw when first starting a mission for example.

Reply:
ok thanks for the feedback.  I've changed this line so that if it fails to get the yaw target from the position controller it falls back to the current attitude controller target.

I momentarily considered changing the position controller so that get_yaw_cd always succeeds and in cases where it would fail it simply returns the current attitude controller yaw target.. That would also work but.. perhaps in some way it is better for the caller to know that the target is actually the current target?